src/client/Microsoft.Identity.Client/Platforms/Android/AndroidBrokerHelper.cs

```diff
@@ -159,7 +166,14 @@ public async Task<string> GetBrokerAuthTokenSilentlyAsync(IDictionary<string, st
 
         public void GetBrokerAccountInfo(IDictionary<string, string> brokerPayload, Activity callerActivity)
         {
-            var loginHint = GetValueFromBrokerPayload(brokerPayload, BrokerParameter.LoginHint);
+            string loginHint = string.Empty;
+            if (string.IsNullOrEmpty(GetValueFromBrokerPayload(brokerPayload, BrokerParameter.LoginHint)))
+            {
+                _logger.Error(MsalErrorMessage.NoUPNOrAccountIDForSilentAuth);
+                throw new MsalClientException(MsalError.NoUsernameOrAccountIDProvidedForSilentAndroidBrokerAuthentication, MsalErrorMessage.NoUPNOrAccountIDForSilentAuth);
+            }
+
+            loginHint = GetValueFromBrokerPayload(brokerPayload, BrokerParameter.LoginHint);
```

Comment:
you're getting this value twice...maybe move this earlier and if loginHint is null, then go through the if statement above, so you're not doing this twice if there is a login hint provided.